src/main/java/org/springframework/data/couchbase/repository/support/SimpleRxJavaCouchbaseRepository.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.couchbase.repository.support;
+
+import java.io.Serializable;
+
+import com.couchbase.client.java.document.json.JsonArray;
+import com.couchbase.client.java.error.DocumentDoesNotExistException;
+import com.couchbase.client.java.view.AsyncViewResult;
+import com.couchbase.client.java.view.AsyncViewRow;
+import com.couchbase.client.java.view.ViewQuery;
+
+import org.springframework.data.couchbase.core.RxJavaCouchbaseOperations;
+import org.springframework.data.couchbase.core.query.View;
+import org.springframework.data.couchbase.repository.RxJava1CouchbaseRepository;
+import org.springframework.data.couchbase.repository.query.CouchbaseEntityInformation;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import rx.Completable;
+import rx.Observable;
+import rx.Single;
+
+/**
+ * Reactive repository base implementation for Couchbase.
+ *
+ * @author Subhashni Balakrishnan
+ */
+public class SimpleRxJavaCouchbaseRepository<T, ID extends Serializable> implements RxJava1CouchbaseRepository<T, ID> {
```

Comment:
See above. You should decide on one library to implement your reactive support (RxJava1 seems the natural choice). Adoption into Project Reactor types (and vice versa) is handled by the repository layer in Spring Data Commons.